### PR TITLE
Updated Parameter Object of roundRect()

### LINKED ISSUE #22408: roundRect()'s radii can be a DOMPointInit
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/roundRect

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/roundRect#parameters

### What information was incorrect, unhelpful, or incomplete?

The `radii` parameter is said to be either a number or a list.

### What did you expect to see?

The `radii` parameter can also be a `DOMPointInit` object (basically any object with `{x, y}` properties) or a list of such objects, or a list mixing numbers and such objects.  
Passing such an object as radii (or in the list) will modify the arc drawing the rounded rectangle. I'm not sure how this should be worded though.

### Do you have any supporting links, references, or citations?

https://html.spec.whatwg.org/multipage/canvas.html#2dcontext:dom-context-2d-roundrect
> ```idl
> optional (unrestricted double or DOMPointInitor sequence<(unrestricted double or DOMPointInit)>) radii = 0
> ```

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/canvasrenderingcontext2d/roundrect`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/roundRect
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md
* Last commit: https://github.com/mdn/content/commit/b4ab53fbe1501dcd6d7691e81aa67b459f995bce
* Document last modified: 2022-09-22T04:14:06.000Z

</details>

### PR DESCRIPTION
**Pull request by dtrannam**
Fix https://github.com/mdn/content/issues/22408

Added DOMPointInit object as a parameter option for radii parameter.

### CODE REVIEW

File: files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md [hunk 1]
```diff
@@ -50,6 +50,9 @@ roundRect(x, y, width, height, radii)
     If `width` is _negative_ the rounded rectangle is flipped horizontally, so the radius values that normally apply to the left corners are used on the right and vice versa.
     Similarly, when `height` is negative, the rounded rect is flipped vertically.
     The specified radii may be scaled (reduced) if any of the edges are shorter than the combined radius of the vertices on either end.
+    
+    The `radii` parameter can also be a [`DOMPointInit`](en-US/docs/Web/API/DOMPoint/fromPoint) object (basically any object with {x, y} properties) or a list of such objects, or a list mixing numbers and such objects.
```
Comment (Kaiido) [inline files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md:54]: The `DOMPointInit` article has been removed in https://github.com/mdn/content/pull/8788/ so **it seems** the *correct* way of writing that would imply something like 
```md
    The `radii` parameter can also be a {{domxref("DOMPoint")}} or {{domxref("DOMPointReadOnly")}} instance, or an object containing the same properties, or a list of such objects, or a list mixing numbers and such objects.
```
However I'm not sure this is even the right place to insert that. I'd have expected it at l41.

File: files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md [hunk 2]
```diff
@@ -50,6 +50,9 @@ roundRect(x, y, width, height, radii)
...
+    - `{x: 0, y: 0}`
```
Comment (Kaiido) [inline files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md:55]: I doubt this is needed.

**Review by Kaiido — COMMENTED** [review]
Not an authoritative review, so you may want to wait for other reviewers before applying any changes, but a few editorial notes.  
Also I'd have expected a few words on how the radii is affected by the DOMPoint parameters.

File: files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/canvasrenderingcontext2d/roundrect/index.md:55]: `DOMPointInit` is a dictionary, so we should at least not use the name (it is irrelevant for the writer and can be changed in the spec at any time without impact on the users).

**Review by Elchi3 — APPROVED** [review]
I've applied the suggestions. Thanks for your PR! 👍

**Pull request merged**